### DISCUSSION TITLE
Emails not sent - Error: Could not find outbox folder to deliver message

### DISCUSSION THREAD

**Discussion by branestawm2** [Extended-Thunder/send-later#681] [Q&A] [answered]
After recent upgrades to Thunderbird 102.12.0 and SendLater 10.3.6, emails are no longer sent from my Drafts folder by SendLater.

TB error console shows repeated error messages:  "Could not find outbox folder to deliver message".

I found this console script to check for my mailboxes in a previous bug discussion; this topic is now closed presumably resolved:

function listFolders(f) { console.log(`  ${f.type}: ${f.path}`); f.subFolders.map(listFolders); }; var ext = window.ExtensionParent.GlobalManager.extensionMap.get("sendlater3@kamens.us"); var apiCan = [...ext.views][0].apiCan; apiCan.findAPIPath("accounts").list().then(accts => accts.map(a => { console.log(a.name); a.folders.map(listFolders); }));
GMX debugger eval code:1:305

My mailboxes on gmx.com were found:

  inbox: /INBOX debugger eval code:1:35
  drafts: /Drafts debugger eval code:1:35
  templates: /Templates debugger eval code:1:35
  sent: /Sent debugger eval code:1:35

(plus a whole bunch of subfolders in archives: /Archives )

So what am I missing here? Does SendLater now require a mailbox called "outbox" to send emails? If so, local or remote?

TB sends immediate email normally. TB is running on  OpenSuse LEAP 15.5.

Any thoughts/suggestions gratefully received. Thanks in advance.

**Comment by jikamens** [comment, marked as answer]
Send Later has always required an Outbox. See the [user guide](https://extended-thunder.github.io/send-later/) for additional information.

If you don't have an Outbox folder under Local Folders then your profile is broken and I don't know how to fix that so you may need to recreate your profile from scratch.

If you have an Outbox under Local Folders then we can investigate why Send Later can't find it.

One test to try is disabling the Send Later extension and restarting Thunderbird and then composing a message and doing File > Send Later. That should cause Thunderbird to put the message into your Outbox. If that works, then we have a Send Later issue. If that doesn't work, then your profile is broken.

**Reply by branestawm2** [reply]
Thank you for your very prompt response. I didn't have an Outbox in 'Local Folders'. Send Later is working again now.

I didn't need to create a new profile. This is a summary of the steps I took to recreate an outbox for SendLater, based on the helpful troubleshooting Tests 1 thru 4 in the SendLater user guide::

1. Right-click on Local Folders to create a new folder 'Outbox'. Re-start TB. SendLater now finds the outbox and puts emails in it, but they are not sent.
2. Disable SendLater. Restart TB. Mails sent with SendLater from File menu are now put in an auto created outbox named 'Send_Later Outbox' (which is a more helpful folder name), but they are not sent.
3. Verify in Config Editor that mail.default_sendlater_uri points to location of 'Send_Later Outbox'.
4. Exit TB. Delete manually Outbox and Outbox.msf and  'Send_Later Outbox' & 'Send_Later Outbox.msf' files from 'Local Folders' in TB profile folder.
5. Start TB. Enable SendLater. Restart TB. Emails now sent correctly by SendLater.

There may be a more efficient way to fix my issue, but the above worked for me. Hopefully this may help someone else with a similar issue.

Thanks again.

**Reply by jikamens** [reply]
Do you have any idea how you deleted your Outbox in the first place? Also, what was the value of the `default_sendlater_uri` setting before you changed it?
I am unable to delete the Outbox; Thunderbird doesn't let me. I would like to reproduce this issue so I can test solutions and document the most straightforward one, but if I can't figure out how to delete the Outbox I can't reproduce the issue.

**Reply by branestawm2** [reply]
I don't believe that I deleted the Outbox in the first place. I have no explanation for why it went missing, presumably after the upgrades.

I didn't change the default_sendlater_uri setting (I don't mess with Config Editor!). I simply verified that this setting was was correct after auto-creation of 'Send_Later Outbox'.

I didn't delete Outbox and 'Send_Later Outbox' from within Thunderbird while restoring SendLater functionality - I believe that TB prevents this, as you observed. I did it with a file browser after exiting TB.  'Send_Later Outbox' was then auto-recreated on TB restart.

HTH

**Reply by jikamens** [reply]
*sigh* This has happened numerous times over the years, and no one can ever explain how their Outbox folder got deleted.
Don't get me wrong, not blaming you here, at this point I think it's clear that something weird is going on within TB, I just don't know what it is or how to reproduce it so I can't report it to the TB folks as something for them to look at. Bugs like this are a pain.

In any case thanks for documenting how you fixed it.
